src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingService.java

```diff
@@ -426,12 +429,40 @@ public IBinder onBind(Intent intent) {
 
     /**
      * Stops the native binary.
+     *
+     * Introduced by #PR599:
+     * Service may also be stopped by a broadcast-intent
+     * The native binary crashes if it is stopped before it is fully initialized. So some logic is
+     * added to circumvent this situation:
+     * If mCurrentState != State.INIT and != State.STARTING it is safe to shutdown the service. In
+     * this case the old execution path is used and the service immediately shut down.
+     * In the other case PollWebGuiAvailableTaskImpl is active (started from
+     * SyncthingService.updateState()) and polling for the WebGUI to become active. Once that
+     * happens the native binary can be shut down safely. PollWebGuiAvailableTaskImpl.onPostExecute
+     * already did this if mStopScheduled is true.
+     * A problem could arise if onDestroy is called when PollWebGuiAvailableTaskImpl.onPostExecute
+     * is just about to start the service. Then onDestroy sets mStopScheduled too late. and the
+     * binary is left running.
+     * So a lock is introduced:
+     * Either onDestroy is first. Then PollWebGuiAvailableTaskImpl.oonPostExecute waits for onDestroy
+     * to finish -> mStopScheduled is set and the binary is shut down then.
+     * Or PollWebGuiAvailableTaskImpl.onPostExecute is first. Then onDestroy waits and then kills
+     * the now active sycnthing binary on the "old" execution path.
```

Comment:
Too short?

The native binary crashes if stopped before it is fully active. In that case signal the stop
request to PollWebGuiAvailableTaskImpl that is active in that situation and terminate the
service there.